docs/standard/basetypes/classes.md

```diff
@@ -0,0 +1,1213 @@
+
+You can use any character class with character class subtraction. To define the set of characters that consists of all Unicode characters from \u0000 through \uFFFF except white-space characters (**\s**), the characters in the punctuation general category (**\p{P}**), the characters in the **IsGreek** named block (**\p{IsGreek}**), and the Unicode NEXT LINE control character (\x85), use `[\u0000-\uFFFF-[\s\p{P}\p{IsGreek}\x85]]`.
+
+Choose character classes for a character class subtraction expression that will yield useful results. Avoid an expression that yields an empty set of characters, which cannot match anything, or an expression that is equivalent to the original base group. For example, the empty set is the result of the expression `[\p{IsBasicLatin}-[\x00-\x7F]]`, which subtracts all characters in the **IsBasicLatin** character range from the **IsBasicLatin** general category. Similarly, the original base group is the result of the expression `[a-z-[0-9]]`. This is because the base group, which is the character range of letters from "a" through "z", does not contain any characters in the excluded group, which is the character range of decimal digits from "0" through "9". 
+
+The following example defines a regular expression, `^[0-9-[2468]]+$`, that matches zero and odd digits in an input string. The regular expression is interpreted as shown in the following table.
+
+Element | Description
+------- | ----------- 
+`^` | Begin the match at the start of the input string.
+`[0-9-[2468]]+` | Match one or more occurrences of any character from 0 to 9 except for 2, 4, 6, and 8. In other words, match one or more occurrences of zero or an odd digit.
+`$` | End the match at the end of the input string.
+ 
+```csharp
+using System;
+using System.Text.RegularExpressions;
+
+public class Example
+{
+   public static void Main()
+   {
+      string[] inputs = { "123", "13579753", "3557798", "335599901" };
+      string pattern = @"^[0-9-[2468]]+$";
+
+      foreach (string input in inputs)
+      {
+         Match match = Regex.Match(input, pattern);
+         if (match.Success) 
+            Console.WriteLine(match.Value);
+      }      
+   }
+}
+// The example displays the following output:
+//       13579753
+//       335599901
+```
+
+```vb
+Imports System.Text.RegularExpressions
+
+Module Example
+   Public Sub Main()
+      Dim inputs() As String = { "123", "13579753", "3557798", "335599901" }
+      Dim pattern As String = "^[0-9-[2468]]+$"
+
+      For Each input As String In inputs
+         Dim match As Match = Regex.Match(input, pattern)
+         If match.Success Then Console.WriteLine(match.Value)
+      Next
+   End Sub
+End Module
+' The example displays the following output:
+'       13579753
+'       335599901
+```
+
+## See Also
+
+[Regular expression language - quick reference](quickref.md)
+
```

Comment:
I think the See Also section on these topics would look better without these extra new lines in between links.